### ISSUE TITLE
Release 3.0.0

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-react-native#1701]
* Align `event.origin` and `event.environment`
* Introduce `Sentry.initWith` that takes the root component
* Add tracing integration OOTB
* Auto detect routing library (React Navigation v4 or v5 and in Q3, React Native Navigation (wix))

**Comment by Bardiamist** [comment]
Is migration guide exist? Now I tried to update to v3 and can't build app.

```
/Users/bardiamist/Library/Developer/Xcode/DerivedData/proj-gcfwctharjklzxamsqckslyxgkbd/Build/Intermediates.noindex/proj.build/Debug-iphonesimulator/proj.build/Script-00DD1BFF1BD5951E006B06BC.sh: line 4: ../node_modules/@sentry/cli/bin/sentry-cli: No such file or directory
Command PhaseScriptExecution failed with a nonzero exit code
```

**Comment by jennmueng** [comment]
@Bardiamist Try a full node_modules reinstall, that error looks to be a missing sentry cli. There shouldn't be changes needed for 3.0.0 apart from different `event.origin` values.

**Comment by Bardiamist** [comment]
> @Bardiamist Try a full node_modules reinstall, that error looks to be a missing sentry cli. There shouldn't be changes needed for 3.0.0 apart from different `event.origin` values.

Unfortunately not helped

**Comment by christophemenager** [comment]
I do have the same error @jennmueng.
It appears that in 3.0.0 there is no cli folder in the /node_modules/@sentry, so the error looks logical
How could we fix this ?

**Comment by benlondon** [comment]
>  I do have the same error @jennmueng.
It appears that in 3.0.0 there is no cli folder in the /node_modules/@sentry, so the error looks logical
How could we fix this ?

While not the best long term solution, adding an explicit dependency on the CLI worked for me:

```
"dependencies": {
  "@sentry/react-native": "^3.0.0",
  "@sentry/cli": "^1.68.0",
...
```

**Comment by nomi9995** [comment]
> > I do have the same error @jennmueng.
> > It appears that in 3.0.0 there is no cli folder in the /node_modules/@sentry, so the error looks logical
> > How could we fix this ?
> 
> While not the best long term solution, adding an explicit dependency on the CLI worked for me:
> 
> ```
> "dependencies": {
>   "@sentry/react-native": "^3.0.0",
>   "@sentry/cli": "^1.68.0",
> ...
> ```

yes it is working

**Comment by CesarLanderos** [comment]
can confirm that worked for me as well, the new version does not include the CLI as a dependency.

I don't know if versions 2.x.x had it or was part of the wizard to install it magically

**Comment by user512** [comment]
Thank you for all the hard work!
I upgraded to 3.0 and was able to use the workaround from https://github.com/getsentry/sentry-react-native/issues/1701#issuecomment-910231456 to make things work.

In https://github.com/getsentry/sentry-react-native/issues/1701#issuecomment-905623863, it was mentioned there's a migration guide but I don't see it here: https://github.com/getsentry/sentry-react-native/blob/477f6fe42d1810501096ec1ecf5308ad5f0530d4/CHANGELOG.md, would you mind pointing me to the right direction?

**Comment by jennmueng** [comment]
@user512 We ended up not needing a full migration guide as we ended up making it fully backward compatible apart from the `event.origin` tag values which we are working on documenting soon.

**Comment by jennmueng** [comment]
Closing this issue now. The `sentry-cli` missing dependency bug is now fixed in version [3.0.1](https://github.com/getsentry/sentry-react-native/releases/tag/3.0.1).

**There is no migration guide** as no changes will need to be made in your codebase when upgrading to 3.x from 2.x.
